src/main/java/amazin/controller/BookController.java

```diff
@@ -65,15 +68,29 @@ public String updateBook(@PathVariable("id") Long id, @Valid Book book, BindingR
             book.setId(id);
             return VIEW_UPDATE_BOOK;
         }
-
         bookService.update(book);
         return "redirect:/";
     }
 
     @GetMapping("delete/{id}")
     public String deleteBook(@PathVariable("id") Long id, Model model) {
+        Optional<Book> bookToBeDeleted = bookService.findById(id);
+        if (bookToBeDeleted.isPresent()) {
+            Book book = bookToBeDeleted.get();
+            deleteFile(book.getPicture_url());
+        }
         bookService.delete(id);
         return "redirect:/";
     }
 
+    @PostMapping("/storage/uploadFile")
+    public String uploadFile(@RequestPart(value = "file") MultipartFile file) {
+        return this.amazonService.uploadFile(file);
+    }
+
+    @DeleteMapping("/storage/deleteFile")
+    public String deleteFile(@RequestPart(value = "url") String fileUrl) {
```

Comment:
do we want users to be able to delete pictures for books without deleting the post for the book?

Reply:
I agree with Aaron, this should be a private method that we call when the book is deleted

Reply:
this is only called when the book is being deleted but good point on the private part